src/fragments/lib/datastore/native_common/advanced-workflows.mdx

```diff
@@ -0,0 +1,272 @@
+<th>
+Autopopulate with UUID
+</th>
+</tr>
+<tr>
+<td>
+no @primaryKey
+</td>
+<td>
+TODO: render type in td
+</td>
+<td>
+Yes
+</td>
+</tr>
+<tr>
+<td>
+no @primaryKey, explicit `id` field
+</td>
+<td>
+TODO: render type in td
+</td>
+<td>
+Yes
+</td>
+</tr>
+<tr>
+<td>
+@primaryKey on a custom field
+</td>
+<td>
+TODO: render type in td
+</td>
+<td>
+No
+</td>
+</tr>
+<tr>
+<td>
+explicit @primaryKey on `id` field
+</td>
+<td>
+TODO: render type in td
+</td>
+<td>
+If not provided in constructor/builder
+</td>
+</tr>
+<tr>
+<td>
+explicit @primaryKey on `id` field with sort key
+</td>
+<td>
+TODO: render type in td
+</td>
+<td>
+If not provided in constructor/builder
+</td>
+</tr>
+<tr>
+<td>
+explicit `id` field in sort key
+</td>
+<td>
+TODO: render type in td
+</td>
+<td>
+If not provided in constructor/builder
+</td>
+</tr>
+<tr>
+<td>
+@primaryKey with no `id` field
+</td>
+<td>
+TODO: render type in td
+</td>
+<td>
+No
+</td>
+</tr>
+</tbody>
+</table>
+
+### Querying records
+
+import android0 from "/src/fragments/lib/datastore/android/advanced-workflows/custom-primary-key/query-snippet.mdx";
+
+<Fragments fragments={{android: android0}} />
+
+import flutter0 from "/src/fragments/lib/datastore/flutter/advanced-workflows/custom-primary-key/query-snippet.mdx";
+
+<Fragments fragments={{flutter: flutter0}} />
+
+import ios0 from "/src/fragments/lib/datastore/ios/advanced-workflows/custom-primary-key/query-snippet.mdx";
+
+<Fragments fragments={{ios: ios0}} />
+
+import js0 from "/src/fragments/lib/datastore/js/advanced-workflows/custom-primary-key/query-snippet.mdx";
+
+<Fragments fragments={{js: js0}} />
+
+
+### Deleting records
+
+import android1 from "/src/fragments/lib/datastore/android/advanced-workflows/custom-primary-key/delete-snippet.mdx";
+
+<Fragments fragments={{android: android1}} />
+
+import flutter1 from "/src/fragments/lib/datastore/flutter/advanced-workflows/custom-primary-key/delete-snippet.mdx";
+
+<Fragments fragments={{flutter: flutter1}} />
+
+import ios1 from "/src/fragments/lib/datastore/ios/advanced-workflows/custom-primary-key/delete-snippet.mdx";
+
+<Fragments fragments={{ios: ios1}} />
+
+import js1 from "/src/fragments/lib/datastore/js/advanced-workflows/custom-primary-key/delete-snippet.mdx";
+
+<Fragments fragments={{js: js1}} />
+
+### Many-to-many
+
+A schema using custom primary keys with a `many-to-many` relationship is converted into a `hasMany` / `belongsTo` relationship with a join table.
+
+For instance, with the following schema:
+```graphql
+type Post @model {
+	customPostId: ID! @primaryKey(sortKeyFields: ["title"])
+	title: String!
+	content: String
+	tags: [Tag] @manyToMany(relationName: "PostTags")
+}
+
+type Tag @model {
+	customTagId: ID! @primaryKey(sortKeyFields: ["label"])
+	label: String!
+	posts: [Post] @manyToMany(relationName: "PostTags")
+}
+```
+
+Codegen will generate the following models (NOTE: if we keep this section, we will split this up and create snippets for each platform):
```

Comment:
NOTE: if we keep this section, we will split this up and create snippets for each platform